flink-java/src/test/java/org/apache/flink/api/java/hadoop/mapred/HadoopOutputFormatTest.java

```diff
@@ -0,0 +1,216 @@
+        JobConf jobConf = mock(JobConf.class);
+
+        HadoopOutputFormat<String, Long> outputFormat = new HadoopOutputFormat<>(dummyOutputFormat, jobConf);
+
+        outputFormat.configure(Matchers.<org.apache.flink.configuration.Configuration>any());
+
+        verify(dummyOutputFormat, times(1)).configure(any(JobConf.class));
+    }
+
+    @Test
+    public void testCloseWithTaskCommit() throws Exception {
+        OutputFormat<String, Long> dummyOutputFormat = mock(DummyOutputFormat.class);
+        DummyOutputCommitter outputCommitter = mock(DummyOutputCommitter.class);
+        when(outputCommitter.needsTaskCommit(any(TaskAttemptContext.class))).thenReturn(true);
+        DummyRecordWriter recordWriter = mock(DummyRecordWriter.class);
+        JobConf jobConf = mock(JobConf.class);
+
+        HadoopOutputFormat<String, Long> outputFormat = new HadoopOutputFormat<>(dummyOutputFormat, jobConf);
+        outputFormat.recordWriter = recordWriter;
+        outputFormat.outputCommitter = outputCommitter;
+
+        outputFormat.close();
+
+        verify(recordWriter, times(1)).close(any(Reporter.class));
+        verify(outputCommitter, times(1)).commitTask(any(TaskAttemptContext.class));
+    }
+
+    @Test
+    public void testCloseWithoutTaskCommit() throws Exception {
+        OutputFormat<String, Long> dummyOutputFormat = mock(DummyOutputFormat.class);
+        DummyOutputCommitter outputCommitter = mock(DummyOutputCommitter.class);
+        when(outputCommitter.needsTaskCommit(any(TaskAttemptContext.class))).thenReturn(false);
+        DummyRecordWriter recordWriter = mock(DummyRecordWriter.class);
+        JobConf jobConf = mock(JobConf.class);
+
+        HadoopOutputFormat<String, Long> outputFormat = new HadoopOutputFormat<>(dummyOutputFormat, jobConf);
+        outputFormat.recordWriter = recordWriter;
+        outputFormat.outputCommitter = outputCommitter;
+
+        outputFormat.close();
+
+        verify(recordWriter, times(1)).close(any(Reporter.class));
+        verify(outputCommitter, times(0)).commitTask(any(TaskAttemptContext.class));
+    }
+
+    @Test
+    public void testWriteRecord() throws Exception {
+        OutputFormat<String, Long> dummyOutputFormat = mock(DummyOutputFormat.class);
+        DummyRecordWriter recordWriter = mock(DummyRecordWriter.class);
+        JobConf jobConf = mock(JobConf.class);
+
+        HadoopOutputFormat<String, Long> outputFormat = new HadoopOutputFormat<>(dummyOutputFormat, jobConf);
+        outputFormat.recordWriter = recordWriter;
+
+        outputFormat.writeRecord(new Tuple2<>("key", 1L));
+
+        verify(recordWriter, times(1)).write(anyString(), anyLong());
+    }
+
+    @Test
+    public void testFinalizeGlobal() throws Exception {
+        OutputFormat<String, Long> dummyOutputFormat = mock(DummyOutputFormat.class);
+        DummyOutputCommitter outputCommitter = mock(DummyOutputCommitter.class);
+        JobConf jobConf = mock(JobConf.class);
```

Comment:
This line needs to be replaced by `JobConf jobConf = spy(new JobConf());`.

The Hadoop code copies the `JobConf` and access additional fields that cannot be mocked.